test/unit/ui.js

```diff
@@ -423,5 +427,123 @@ describe('ui class', () => {
+          ui.components.modal = [{options: {}}];
+          ui.createModal({options: {}});
+          assert.equal(1, ui.components.modal.length);
+        });
+
+        it('returns the newly created modal', () => {
+          assert.instanceOf(ui.createModal({options: {}}), Modal);
+        });
+
+        it('returns the first modal if a modal already exists', () => {
+          ui.components.modal = ['test'];
+          assert.equal(ui.createModal({options: {}}), 'test');
+        });
+      });
+
+      describe('closeModal()', () => {
+        let restoreFocusStub;
+
+        beforeEach(() => {
+          restoreFocusStub = sinon.stub(UI.prototype, 'restoreFocus');
+        });
+
+        afterEach(() => {
+          restoreFocusStub.restore();
+        });
+
+        it('closes every modal then restores focus once all is done', () => {
+          const closeSpy1 = sinon.spy();
+          const closeSpy2 = sinon.spy();
+          const closeSpy3 = sinon.spy();
+          ui.components.modal = [
+            {close: closeSpy1},
+            {close: closeSpy2},
+            {close: closeSpy3},
+          ];
+
+          ui.closeModal();
+          assert.calledOnce(closeSpy1);
+          assert.calledOnce(closeSpy2);
+          assert.calledOnce(closeSpy3);
+          assert.calledOnce(restoreFocusStub);
+        });
+
+        it('does not restore focus if there are no modals', () => {
+          ui.components.modal = [];
+          ui.closeModal();
+          assert.notCalled(restoreFocusStub);
+        });
+      });
+    });
+
+    describe('setActiveEl()', () => {
+      it('sets activeEl to element param', () => {
+        ui.setActiveEl('test');
+        assert.equal(ui.activeEl, 'test');
+      });
+    });
+
+    describe('restoreFocus()', () => {
+      let focusSpy;
+
+      beforeEach(() => {
+        focusSpy = sinon.spy();
+        ui.activeEl = {focus: focusSpy};
+      });
+
+      it('focuses on activeEl if activeEl exists and both modal and cart are closed', () => {
+        ui = Object.defineProperty(ui, 'modalOpen', {
+          value: false,
+          writable: false,
+        });
+        ui = Object.defineProperty(ui, 'cartOpen', {
+          value: false,
+          writable: false,
+        });
+
+        ui.restoreFocus();
+        assert.calledOnce(focusSpy);
+      });
+
+      it('does not focus on activeEl if modal is open', () => {
+        ui = Object.defineProperty(ui, 'modalOpen', {
+          value: true,
+          writable: false,
+        });
+        ui = Object.defineProperty(ui, 'cartOpen', {
+          value: false,
+          writable: false,
+        });
+
+        ui.restoreFocus();
+        assert.notCalled(focusSpy);
+      });
+
+      it('does not focus on activeEl if cart is open', () => {
+        ui = Object.defineProperty(ui, 'modalOpen', {
+          value: false,
+          writable: false,
+        });
+        ui = Object.defineProperty(ui, 'cartOpen', {
```

Comment:
Could we do the `Object.defineProperty` inside the `beforeEach` and only set `writable: true` without a value? This would clean up the subsequent tests because we could plainly set the values using `ui.modalOpen = true`